proposals/dao/tip_121a.ts

```diff
@@ -0,0 +1,204 @@
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  // display pcvStats
+  console.log('----------------------------------------------------');
+  console.log(' pcvStatsBefore.protocolControlledValue [M]e18 ', Number(pcvStatsBefore.protocolControlledValue) / 1e24);
+  console.log(' pcvStatsBefore.userCirculatingFei      [M]e18 ', Number(pcvStatsBefore.userCirculatingFei) / 1e24);
+  console.log(' pcvStatsBefore.protocolEquity          [M]e18 ', Number(pcvStatsBefore.protocolEquity) / 1e24);
+  const pcvStatsAfter: PcvStats = await contracts.collateralizationOracle.pcvStats();
+  console.log('----------------------------------------------------');
+  console.log(' pcvStatsAfter.protocolControlledValue  [M]e18 ', Number(pcvStatsAfter.protocolControlledValue) / 1e24);
+  console.log(' pcvStatsAfter.userCirculatingFei       [M]e18 ', Number(pcvStatsAfter.userCirculatingFei) / 1e24);
+  console.log(' pcvStatsAfter.protocolEquity           [M]e18 ', Number(pcvStatsAfter.protocolEquity) / 1e24);
+  console.log('----------------------------------------------------');
+  const pcvDiff = pcvStatsAfter.protocolControlledValue.sub(pcvStatsBefore.protocolControlledValue);
+  const cFeiDiff = pcvStatsAfter.userCirculatingFei.sub(pcvStatsBefore.userCirculatingFei);
+  const eqDiff = pcvStatsAfter.protocolEquity.sub(pcvStatsBefore.protocolEquity);
+  console.log(' PCV diff                               [M]e18 ', Number(pcvDiff) / 1e24);
+  console.log(' Circ FEI diff                          [M]e18 ', Number(cFeiDiff) / 1e24);
+  console.log(' Equity diff                            [M]e18 ', Number(eqDiff) / 1e24);
+  console.log('----------------------------------------------------');
+
+  // check amount of TRIBE returned in DAO treasury from buybacks pool
+  const tribeInDaoTreasuryAfter = await contracts.tribe.balanceOf(addresses.core);
+  const tribeReturnedToDaoTreasury = tribeInDaoTreasuryAfter.sub(tribeInDaoTreasuryBefore);
+  expect(tribeReturnedToDaoTreasury).to.be.at.least(e18('2200000')); // 2.2M TRIBE
+
+  // Check lenses & balances
+  // LUSD -> DAI pool lenses
+  expect((await contracts.lusdToDaiLensDai.resistantBalanceAndFei())[0]).to.be.at.least(e18('1000000')); // 1M DAI
+  expect((await contracts.lusdToDaiLensDai.resistantBalanceAndFei())[1]).to.be.equal('0'); // 0 FEI
+  expect((await contracts.lusdToDaiLensLusd.resistantBalanceAndFei())[0]).to.be.at.least(e18('18500000')); // 18.5M LUSD
+  expect((await contracts.lusdToDaiLensLusd.resistantBalanceAndFei())[1]).to.be.equal('0'); // 0 FEI
+  // LUSD -> DAI pool balances
+  const lusdPoolTokens = await contracts.balancerVault.getPoolTokens(poolId);
+  expect(lusdPoolTokens.balances[0]).to.be.at.least(e18('18500000')); // 18.5M LUSD
+  expect(lusdPoolTokens.balances[1]).to.be.at.least(e18('1000000')); // 1M DAI
+  // WETH -> DAI pool balances
```

Comment:
Is it worthwhile performing a swap on the LBP and getting the inferred price out?

Reply:
sure let's do this